packages/rum-core/src/boot/rumPublicApi.ts

```diff
@@ -18,14 +18,18 @@ import {
   InternalMonitoring,
   callMonitored,
   createHandlingStack,
+  Omit,
 } from '@datadog/browser-core'
 import { RumEventDomainContext } from '../domainContext.types'
 import { CommonContext, User, ActionType } from '../rawRumEvent.types'
 import { RumEvent } from '../rumEvent.types'
 import { buildEnv } from './buildEnv'
 import { startRum } from './startRum'
 
-export interface RumInitConfiguration extends InitConfiguration {
+const ignoredConfigurationOptions = { publicApiKey: true, datacenter: true } as const
+type IgnoredConfigurationOptions = keyof typeof ignoredConfigurationOptions
```

Comment:
Nice! It could be better to express this as an array:

```suggestion
const ignoredConfigurationOptions = [ 'publicApiKey', 'datacenter' ] as const
type IgnoredConfigurationOptions = (typeof ignoredConfigurationOptions)[number]
```

Reply:
Sure !

Reply:
Not compatible with Typescript 3.0
Only available for TypeScript 3.4

Reply:
I used a tuple helper for compatibility with Typescript v3.0